sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -317,6 +321,7 @@ def is_valid_split_point(index):
   def finish(self):
     # type: () -> None
     with self.splitting_lock:
+      self.index += 1
```

Comment:
It doesn't seem like the index should be incremented here.

Reply:
We want the final count to represent the number of fully processed elements so when we process 0 elements in the bundle we should end with 0 and not -1. Similarly when we process 1 element in the bundle the final index should be 1 and not 0.